alphastats/DataSet.py

```diff
@@ -66,47 +67,81 @@ def __init__(
         """
         self._check_loader(loader=loader)
 
+        # fill data from loader
         self.rawinput: pd.DataFrame = loader.rawinput
-        self.software: str = loader.software
-        self.index_column: str = loader.index_column
-        self.intensity_column: Union[str, list] = loader.intensity_column
         self.filter_columns: List[str] = loader.filter_columns
-        self.evidence_df: pd.DataFrame = loader.evidence_df
-        self.gene_names: str = loader.gene_names
-
-        # include filtering before
-        self._create_matrix()
-        self._check_matrix_values()
-
-        self.metadata: pd.DataFrame
-        self.sample: str
-        if metadata_path is not None:
-            self.sample = sample_column
-            self.metadata = self._load_metadata(file_path=metadata_path)
-            self._remove_misc_samples_in_metadata()
-        else:
-            self.sample = "sample"
-            self.metadata = pd.DataFrame({"sample": list(self.mat.index)})
-
-        if loader == "Generic":
-            intensity_column = loader._extract_sample_names(
+        self.index_column: str = loader.index_column
+        self.software: str = loader.software
+        self._gene_names: str = loader.gene_names
+
+        self._intensity_column: Union[str, list] = (
```

Comment:
note that this was different before:
`loader.intensity_column` was used to create the `mat` object, and only afterwards overwritten in the case of a generic loader ..